Update documentation for Quarkus distribution (#1385)
keycloak/keycloak-documentation: server_admin/topics/authentication/x509.adoc

* Certificate KeyUsage validation.
* Certificate ExtendedKeyUsage validation.

> ifeval::["{kc_dist}" == "wildfly"]

Author: Follow-up for new TLS guide: https://github.com/keycloak/keycloak/issues/10041

> ==== Enable X.509 client certificate user authentication
> 
> ifeval::["{kc_dist}" == "wildfly"]
> The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.

Author: ```suggestion
ifeval::["{kc_dist}" == "wildfly"]
==== Enable X.509 client certificate user authentication

The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.
```

The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.

> ifeval::["{kc_dist}" == "wildfly"]

Reviewer: We have `https-client-auth`. Do we need an example for Quarkus or that is already covered by TLS guide?

Author: It should be covered in the new TLS guide, and not here. Looking at it now it's not explicitly mentioned, but "https-client-auth" is at least under the relevant options. I think that's sufficient for now, and we can extend on mtls coverage after the 17 release?

> The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.
> 
> ifeval::["{kc_dist}" == "wildfly"]

Author: ```suggestion
ifeval::["{kc_dist}" == "wildfly"]
The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.

```

Defines the method for extracting the user identity from a client certificate.

*Canonical DN representation enabled*::

> Defines whether to use canonical format to determine a distinguished name. The official link:https://docs.oracle.com/javase/8/docs/api/javax/securityauth/x500/X500Principal.html#getName-java.lang.String-[Java API documentation] describes the format. This option affects the two User Identity Sources _Match SubjectDN using regular expression_ and _Match IssuerDN using regular expression_ only. Enable this option when you set up a new {project_name} instance. Disable this option to retain backward compatibility with existing {project_name} instances.

Author: ```suggestion
Defines whether to use canonical format to determine a distinguished name. The official link:https://docs.oracle.com/javase/8/docs/api/javax/security/auth/x500/X500Principal.html#getName-java.lang.String-[Java API documentation] describes the format. This option affects the two User Identity Sources _Match SubjectDN using regular expression_ and _Match IssuerDN using regular expression_ only. Enable this option when you set up a new {project_name} instance. Disable this option to retain backward compatibility with existing {project_name} instances.
```

{project_name} does not have built-in support for other reverse proxy implementations. However, you can make other reverse proxies behave in a similar way to `apache` or `haproxy`. If none of these work, create your implementation of the `org.keycloak.services.x509.X509ClientCertificateLookupFactory` and `org.keycloak.services.x509.X509ClientCertificateLookup` providers. See the link:{developerguide_link}[{developerguide_name}] for details on how to add your provider.

> ifeval::["{kc_dist}" == "wildfly"]

Reviewer: Except for the `Dumping Headers` section, it should be possible to enable logging and follow the troubleshooting steps.

Author: We should rather add a troubleshooting guide in the new server guides, and link to that. I would consider that as a post-17 follow-up task though
